src/App.tsx

```diff
@@ -2,18 +2,156 @@ import React from 'react';
 import './App.css';
 
 import users from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
-};
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: users.find(person => person.id === todo.userId) || null,
+}));
+
+interface State {
+  user: string;
+  title: string;
+  todoList: Todo[];
+  isUserEmpty: boolean;
+  isTitleEmpty: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    user: '',
+    title: '',
+    todoList: preparedTodos,
+    isUserEmpty: false,
+    isTitleEmpty: false,
+  };
+
+  handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event.target.value.replace(/[^ a-zA-Zа-яА-я0-9]/g, ''),
+      isTitleEmpty: false,
+    });
+  };
+
+  selectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      user: event.target.value,
+      isUserEmpty: false,
+    });
+  };
+
+  addTodo = () => {
+    const { todoList, title, user } = this.state;
+
+    const currentUser: User | null = users.find(
+      (userFromServer) => userFromServer.name === user,
+    ) || null;
+
+    if (!currentUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      user: currentUser,
+      userId: currentUser.id,
+      id: todoList[todoList.length - 1].id + 1,
+      title,
+      completed: false,
+    };
+
+    this.setState((state) => ({
+      todoList: [...state.todoList, newTodo],
+    }));
+  };
+
+  validateSubmit = () => {
+    const { title, user } = this.state;
+
+    this.setState({
+      isTitleEmpty: title === '',
+      isUserEmpty: user === '',
+    });
+
+    return title && user;
+  };
+
+  clearForm = () => {
+    this.setState({
+      user: '',
+      title: '',
+    });
+  };
+
+  render() {
+    const {
+      user, title, todoList, isUserEmpty, isTitleEmpty,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <div className="todos text-center">
+          <h1 className="mb-5">Add todo form</h1>
+          <form
+            onSubmit={(event) => {
+              event.preventDefault();
+              this.validateSubmit();
```

Comment:
your validateSubmit method return form valid value, but you don't catch this
don't call addTodo and clearForm if form inputs don’t valid